cmd/kubeadm/app/cmd/token.go

```diff
@@ -85,7 +92,7 @@ func NewCmdToken(out io.Writer, errW io.Writer) *cobra.Command {
 	}
 
 	tokenCmd.PersistentFlags().StringVar(&kubeConfigFile,
-		"kubeconfig", "/etc/kubernetes/admin.conf", "The KubeConfig file to use when talking to the cluster")
+		"kubeconfig", defaultKubeConfig, "The KubeConfig file to use when talking to the cluster. If the flag is not set the environment variable "+kubeConfigEnv+" and the user home directory and will be searched. If no existing KubeConfig files are found there the default location "+defaultKubeConfig+" will be used.")
```

Comment:
>If the flag is not set the environment variable "+kubeConfigEnv+" and the user home directory and will be searched.

I don't understand this. Guess the last `and` is extra?

>If no existing KubeConfig files are found there the default location "+defaultKubeConfig+" will be used."

I'm not sure if `defaultKubeConfig` can be printed in this format. Let's see if the CI will be happy.

/ok-to-test

Reply:
yes, there is an extra `and` - typo.
i will probably replace this comment based on what `clientcmd.NewNonInteractiveDeferredLoadingClientConfig` (what a name...) does exactly behind the lines. on a quick look the `cmdutil.GetJoinCommand` call might have a problem with that...